src/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.cs

```diff
@@ -28,23 +28,92 @@ public DependencyContext Read(Stream stream)
             var targetsObject = (IEnumerable<KeyValuePair<string, JToken>>) root[DependencyContextStrings.TargetsPropertyName];
 
-            var runtimeTargetProperty = targetsObject.First(target => IsRuntimeTarget(target.Key));
-            var compileTargetProperty = targetsObject.First(target => !IsRuntimeTarget(target.Key));
+            JObject runtimeTarget = null;
+            JObject compileTarget = null;
+            if (targetsObject != null)
+            {
+                var compileTargetProperty = targetsObject.FirstOrDefault(t => !IsRuntimeTarget(t.Key));
+                compileTarget = (JObject) compileTargetProperty.Value;
+                target = compileTargetProperty.Key;
+
+                if (!string.IsNullOrEmpty(runtimeTargetInfo.Name))
+                {
+                    runtimeTarget = (JObject) targetsObject.FirstOrDefault(t => t.Key == runtimeTargetInfo.Name).Value;
+                    if (runtimeTarget == null)
+                    {
+                        throw new FormatException($"Target with name {runtimeTargetInfo.Name} not found");
+                    }
+
+                    var seperatorIndex = runtimeTargetInfo.Name.IndexOf(DependencyContextStrings.VersionSeperator);
+                    if (seperatorIndex > -1 && seperatorIndex < runtimeTargetInfo.Name.Length)
+                    {
+                        runtime = runtimeTargetInfo.Name.Substring(seperatorIndex + 1);
+                    }
+                }
+                else
+                {
+                    runtimeTarget = compileTarget;
+                }
+            }
 
             return new DependencyContext(
-                compileTargetProperty.Key,
-                runtimeTargetProperty.Key.Substring(compileTargetProperty.Key.Length + 1),
+                target,
+                runtime,
+                runtimeTargetInfo.Portable,
                 ReadCompilationOptions((JObject)root[DependencyContextStrings.CompilationOptionsPropertName]),
-                ReadLibraries((JObject)compileTargetProperty.Value, false, libraryStubs).Cast<CompilationLibrary>().ToArray(),
-                ReadLibraries((JObject)runtimeTargetProperty.Value, true, libraryStubs).Cast<RuntimeLibrary>().ToArray()
+                ReadLibraries(compileTarget, false, libraryStubs).Cast<CompilationLibrary>().ToArray(),
+                ReadLibraries(runtimeTarget, true, libraryStubs).Cast<RuntimeLibrary>().ToArray(),
+                ReadRuntimeGraph((JObject)root[DependencyContextStrings.RuntimesPropertyName]).ToArray()
                 );
         }
 
+        private IEnumerable<KeyValuePair<string, string[]>> ReadRuntimeGraph(JObject runtimes)
+        {
+            if (runtimes == null)
+            {
+                yield break;
+            }
+
+            var targets = runtimes.Children();
+            var runtime = (JProperty)targets.Single();
+            foreach (var pair in (JObject)runtime.Value)
```

Comment:
`.ToDictionary`? Since a dictionary is `IEnumerable<KeyValuePair<K, V>>` you can just return that.